accepted/00000-use-dnf-for-rpm-repos.md

```diff
@@ -0,0 +1,177 @@
+  + Alternatively, the `python3-urlgrabber` library ([OBS link](https://build.opensuse.org/package/show/openSUSE:Factory/python-urlgrabber)) could be used.
+
+#### DNF Python code examples
+- Some DNF use cases from Python: https://dnf.readthedocs.io/en/latest/use_cases.html
+- Working with DNF: http://abregman.com/2016/11/29/python-working-with-rpms/
+- DNF `reposync` subcommand code: https://github.com/rpm-software-management/dnf-plugins-core/blob/master/plugins/reposync.py
+  + This implementation will download all metadata referenced by `repomd.xml`, even stuff not currently handled by DNF itself.
+- Example of resolving dependencies in a repo with DNF: https://pagure.io/releng/blob/master/f/scripts/spam-o-matic
+- Some Python examples with librepo: https://github.com/rpm-software-management/librepo/tree/master/examples/python
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main drawback of this would be imposing the DNF stack as a dependency for the server.
+This is only slightly an issue in SUSE Linux Enterprise, as it's maintained in PackageHub.
+For the downstream SUSE Manager product, it is likely that the DNF stack packages would need
+to be imported to be maintained exclusively for being used with SUSE Manager.
+
+However, this is unlikely to be much of a maintenance burden as the community packaging
+is actively maintained and well-structured. The packager (who is also the author of this RFC)
+is responsive to Uyuni/SUSE Manager needs and the upstream DNF developers are interested
+and responsive to community consumers' needs.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Revert back to using YUM
+
+One alternative to resolve this issue is to roll back to using YUM. However, this is a bad idea.
+YUM is dead, and DNF is the maintained successor. YUM does not support Python 3, and there is an
+immediate goal to be able to run Uyuni on a system lacking Python 2.
+
+And even with reverting back to YUM, Uyuni would not be able to properly support RH/Fedora modules.
+Even Spacewalk (which has not yet migrated to DNF) is not able to deal with modules properly as
+they are offered in Red Hat Enterprise Linux 8.
+
+Moreover, [YUM has been retired in Fedora 31](https://fedoraproject.org/wiki/Changes/Retire_YUM_3)
+and "YUM" in Red Hat Enterprise Linux 8 is just a CLI shim to DNF, and only the DNF Python interface
+and libraries are available. Thus, reverting to the "YUM" version would not work.
+
+## Package and port Zypper to other distributions for Uyuni
+
+Another alternative is to package Zypper for distributions that lack it. But this is not appropriate
+for CentOS/RHEL as doing so requires replacing and upgrading packages in the distribution base to
+build it. Notably, the `libsolv` package is too old in CentOS/RHEL, and even if it was not, it is
```

Comment:
Ah, okay. I think this answers my question from above. So libsolv is the main problem here.

Reply:
Ok, it's clear that bringing Zypper to CentOS/RHEL is not possible at the moment without breaking terms of supportability on the RHEL side